finite-state-machine/src/main/java/com/iluwatar/finite/state/machine/states/CorrectFirstLetterState.java

```diff
@@ -0,0 +1,87 @@
+package com.iluwatar.finite.state.machine.states;
+
+import com.iluwatar.finite.state.machine.AutomatonInterfaceI;
+import com.iluwatar.finite.state.machine.DataModel;
+import com.iluwatar.finite.state.machine.Event;
+import com.iluwatar.finite.state.machine.EventSink;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * State when first letter was correct.
+ */
+@Slf4j
+public class CorrectFirstLetterState implements AutomatonInterfaceI {
+
+  /**
+   * Event sink for sending state changes.
+   * */
+  private final EventSink eventSink;
+
+  /**
+   * Event sink for current inputs value.
+   * */
+  private final DataModel model;
+
+
+  /**
+   * Constructor.
+   *
+   * @param newEventSink -> evenSink value.
+   * @param newModel -> model value.
+   * */
+  public CorrectFirstLetterState(
+      final EventSink newEventSink,
+      final DataModel newModel
+  ) {
+    this.eventSink = newEventSink;
+    this.model = newModel;
+  }
+
+  /**
+   * Restarts the FSM.
+   * */
+  @Override
+  public void startNewQuery() {
+    LOGGER.info("Input cache cleared");
+    eventSink.castEvent(Event.CLEAR);
+    model.clear();
+  }
+
+  /**
+   * Input new character into FSM.
+   *
+   * @param character -> value of inputted character.
+   * */
+  @Override
+  public void inputCharacter(final char character) {
+    // check if character is a lowercase letter
+    if ((int) 'a' <= (int) character && (int) character <= (int) 'z') {
```

Comment:
Instead of the primitive type, we could consider using `Character` https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Character.html